pkg/api/user.go

```diff
@@ -33,7 +33,7 @@ func getUserUserProfile(userId int64) Response {
 
 // GET /api/users/loginoremail/:loginOrEmail
 func GetUserByLoginOrEmail(c *middleware.Context) Response {
-	query := m.GetUserByLoginQuery{LoginOrEmail: c.Params(":loginOrEmail")}
+	query := m.GetUserByLoginQuery{LoginOrEmail: c.QueryEscape("loginOrEmail")}
```

Comment:
Thanks for the reply @torkelo . 
> What case does it not return correct results?
I had in mind the email tagging technique (there might be other more complicated use cases) which uses the '+' character in local-part but I double checked it with all the available functions we've discussed so far and it seems to work (with the appropriate encoding). 
> But this is a GET request so not sure that applies here.
I was meant to say that it has an extra logic we don't need/use in this case. But I agree that it is safer not to use the `QueryEscape()`. I have updated the PR to use the `Query()` instead. Let me know if you agree. :)

2. Yeah.. I had in mind https and browser history (parameters will be present in the browser history when inserted via the address bar) when I wrote this. Apparently it is not the case at the moment.